File: js/suitcase.js
"use strict";

// Port of suitcase.java.

class SuitcaseLevel extends KLevel {
  static NUMCASES = 10;

  constructor(game) {
    super(game);
    this.plot = "Larry ist am Flughafen";
    this.inst = "Click on the bag you think is Larry's";

    this.bagClicked = -1;
    this.r = [];
    this.r_base = 0;
    this.time = 0;
  }

  async loadData() {
    this.time = performance.now();

    this.scene = await loadImage("assets/case_scene.gif");
    this.man = await loadImage("assets/case_man.gif");
    this.acase = await loadImage("assets/case.gif");
    for (let i = 0; i < SuitcaseLevel.NUMCASES; i++) {
      this.r[i] = 2 * Math.PI / SuitcaseLevel.NUMCASES * i;
    }

    this.speed = 100;
    this.start();
  }

  _bagXY(i) {
    return {
      x: 199 - 25 + Math.floor(Math.cos(this.r[i] + this.r_base) * 154),
      y: 160 - 18 + Math.floor(Math.sin(this.r[i] + this.r_base) * 49),
    };
  }

  drawNormal(g) {
    this.r_base += (2 * Math.PI) / 360;
    if (this.r_base > 2 * Math.PI) this.r_base -= 2 * Math.PI;

    for (let i = 0; i < SuitcaseLevel.NUMCASES; i++) {
      if (i !== this.bagClicked) {
        const p = this._bagXY(i);
        g.drawImage(this.acase, p.x, p.y);
      }
    }
    g.drawImage(this.man, 200, 150);

    if (this.bagClicked !== -1) {
      this.stop();
      this.displayText(g, "Hmmm... es ist schwerer zu ihn...");
      this.enableNextLevel();
    }
  }

  drawStatic(g) {
    g.drawImage(this.scene, 0, 0);
  }

  mouseDown(x, y) {
    const now = performance.now();
    for (let i = 0; i < SuitcaseLevel.NUMCASES; i++) {
      const p = this._bagXY(i);
      if (x >= p.x && x < p.x + 50 && y >= p.y && y < p.y + 36 &&
          now > this.time + 500) {
        this.bagClicked = i;
      }
    }
  }
}

File: suitcase.java
import java.awt.*;

public class suitcase extends klevel
{
    int bagClicked = -1;
    Image scene, man, acase;

    final static int NUMCASES = 10;
    double r[] = new double[NUMCASES];
    double r_base = 0;

    long time;

    public suitcase ()
    {
        plot = "Larry ist am Flughafen";
        inst = "Click on the bag you think is Larry's";
    }

    public void loadData ()
    {
        time = System.currentTimeMillis();

        scene = waitImage("case_scene.gif");
        man = waitImage("case_man.gif");
        acase = waitImage("case.gif");
        for (int i=0;i<NUMCASES;i++)
        {
            r[i] = 2d*Math.PI / NUMCASES * i;
        }

        loaded=true;
        speed=100;
        start();
    }

    public void drawNormal (Graphics g)
    {
        r_base += (2d*Math.PI) / 360d;
        if (r_base > 2d*Math.PI) r_base-=2d*Math.PI;

        for (int i=0;i<NUMCASES;i++)
        {
            if (i != bagClicked) g.drawImage(acase,199-25+(int)(Math.cos(r[i]+r_base)*154d),160-18+(int)(Math.sin(r[i]+r_base)*49d),this);
        }
        g.drawImage(man,200,150,this);

        if (bagClicked != -1)
        {
            stop();
            displayText(g,"Hmmm... es ist schwerer zu ihn...");
            enableNextLevel();
        }
    }

    public void drawStatic (Graphics g)
    {
        g.drawImage(scene,0,0,this);
    }

    public boolean mouseDown (Event e, int x, int y)
    {
        long now = System.currentTimeMillis();
        boolean good = false;
        for (int i=0;i<NUMCASES;i++)
        {
            Rectangle sc = new Rectangle(199-25+(int)(Math.cos(r[i]+r_base)*154d),160-18+(int)(Math.sin(r[i]+r_base)*49d),50,36);
            if (sc.inside(x,y) && now>(time+500))
            {
                bagClicked = i;
            }
        }
        return true;

    }

}